### ISSUE TITLE
clear variables from included conf files

### ISSUE THREAD

**Issue by davidmarin** [Yelp/mrjob#1162]
(This is an alternative solution to #1156.)

Currently, you can include other config files into your `mrjob.conf` (see #104). For simple options, (e.g. `ami_version`), you can just override them, but for list-like variables (e.g. `setup`), all you can do is append to lists from the included config file. A similar, suble issue exists for environment variables ending in `PATH` (e.g. `cmdenv`).

We should add a `!clear` YAML tag, which can be used to clear out an option, or values in a dictionary-like option.

```yaml
include: /etc/mrjob.base.conf
runners:
  emr:
    bootstrap: !clear
    cmdenv:
      PATH: !clear /just/this/path
    setup: !clear /run/just/this/command with_args
```

**Comment by davidmarin** [comment]
Implementing this against the `v0.4.x` branch so that Yelp, who requested this, can use it immediately.

**Comment by davidmarin** [comment]
Clearly, there isn't a JSON equivalent of this. If you're serious about config files, you should just use YAML!

**Comment by davidmarin** [comment]
Fixed by #1167.
